.github/workflows/type-checking.yml

```diff
@@ -0,0 +1,62 @@
+name: Static Type Checking
+
+on: [pull_request, workflow_dispatch]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  mypy:
+    name: Static Type Checking
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+
+      matrix:
+        include:
+          # TODO: Add testing for older python and/or numpy versions
+          # - python-version: "3.9"
```

Comment:
While I think that this is a great goal, and I might have suggested it.  Python's type checking system is not the best set up to support this.  For example, if we need different typing for vtk 9.1 and 9.4 because the function signature changes in vtk, I don't think type checking can handle this dependency level detail, or even if it can, creates even more complexity.  I like the idea of this PR to implement a 'like-for-like' switch first, which it is.

Reply:
Yea... it will be challenging to support across NumPy and VTK versions. I'm thinking we could add a mechanism to simply ignore this type check conditionally, e.g. if `TYPE_CHECKING and pv.vtk_version > (9.3)`. This is more a nice-to-have though. TBD if we will support this in the future.

Reply:
Maybe we should remove the TODO and the use of `matrix`? I'm thinking that in the future, if we really want to support this properly, we should probably test using the same matrix used for the unit testing.  And in that case, to avoid having to duplicate the matrix, it might be better to run `mypy` as part of that workflow instead. But that's for a later conversion

Reply:
I removed the matrix in [67015d7](https://github.com/pyvista/pyvista/pull/6738/commits/67015d74d3754a6abd9ed313844229efdf0aa7d9)